src/owner.rs

```diff
@@ -0,0 +1,108 @@
+use std::{mem, slice};
+
+use ndarray::{ArrayBase, Dimension, OwnedRepr};
+use pyo3::class::impl_::{PyClassImpl, ThreadCheckerStub};
+use pyo3::pyclass::PyClass;
+use pyo3::pyclass_slots::PyClassDummySlot;
+use pyo3::type_object::{LazyStaticType, PyTypeInfo};
+use pyo3::{ffi, types::PyAny, PyCell};
+
+use crate::dtype::Element;
+
+pub(crate) struct Owner {
+    ptr: *mut u8,
+    len: usize,
+    cap: usize,
+    drop: unsafe fn(*mut u8, usize, usize),
+}
+
+unsafe impl Send for Owner {}
+
+impl<T: Send> From<Box<[T]>> for Owner {
+    fn from(data: Box<[T]>) -> Self {
+        unsafe fn drop_boxed_slice<T>(ptr: *mut u8, len: usize, _cap: usize) {
+            let _ = Box::from_raw(slice::from_raw_parts_mut(ptr as *mut T, len) as *mut [T]);
+        }
+
+        let ptr = data.as_ptr() as *mut u8;
+        let len = data.len();
+        let cap = 0;
+        let drop = drop_boxed_slice::<T>;
+
+        mem::forget(data);
```

Comment:
I prefer to `Box::into_raw` for readability. We use `mem::forget` for various usages, but `into_raw` has only one purpose.

Reply:
Me too. :-) The problem is that `Box::into_raw` will yield `*mut [T]`, i.e. a fat pointer, for the boxed slice which does not apply to the `Vec<_>` case. Hence I manually deconstructed the boxed slice to be able to handle both types in a more or less uniform way.

Reply:
Ahh OK I got it.

Reply:
https://doc.rust-lang.org/std/primitive.pointer.html#method.as_mut_ptr this would work but it's still unstable too 😅

Reply:
Or we can use `std::slice::from_raw_parts` for `Vec` side, but either approach is not so good, so either is fine for me.

Reply:
Yes, I think the current approach is a reasonable compromise as it handles both types in the same way. I did add another `FIXME` comment w.r.t. `Box::into_raw` though.